src/dataframerow/utils.jl

```diff
@@ -258,6 +258,14 @@ function group_rows(df::AbstractDataFrame, hash::Bool = true, sort::Bool = false
 
     # drop group 1 which contains rows with missings in grouping columns
     if skipmissing
+        s = starts[2]
+        if s > 1
+            N = length(rperm) - s + 1
+            copyto!(rperm, 1, rperm, s, N)
```

Comment:
have you considered using a `view` instead of `copyto!`? I think it should be faster and use less memory.